src/defaults/DefaultLinkWidget.tsx

```diff
@@ -16,6 +16,20 @@ export interface DefaultLinkState {
 	selected: boolean;
 }
 
+const addPointToLink = (event, widget, index: number) => {
+	if (
+		!event.shiftKey &&
+		!widget.props.diagramEngine.isModelLocked(widget.props.link) &&
+		widget.props.link.points.length - 1 <= widget.props.diagramEngine.getMaxNumberPointsPerLink()
+	) {
+		const point = new PointModel(widget.props.link, widget.props.diagramEngine.getRelativeMousePoint(event));
+		point.setSelected(true);
+		widget.forceUpdate();
+		widget.props.link.addPoint(point, index);
+		widget.props.pointAdded(point, event);
+	}
+};
```

Comment:
This logic was duplicated when the first point was added vs subsequent points.  
Now it handles the new condition and it is centralised in a single place.

Reply:
can we move this into the widget class as a static field?

Reply:
Yep, no worries, moved it to the class now.